README.md

```diff
@@ -10,16 +10,56 @@ Nebulaworks Engineering use of [nixpkgs](https://github.com/NixOS/nixpkgs)
 * [Templates](./templates/README.md)
 
 ## Building Nix Configurations
-While there are numerous ways of building Nix configurations, one way to build nix configs in a non-linux environment is leveraging a Docker container. The general command for this approach is:
+While there are numerous ways of building Nix configurations, one way to build nix configs in a non-linux environment is
+leveraging a Docker container. The general command for this approach is:
+
 ```
 docker run --rm -it -v $(git rev-parse --show-toplevel):/build -w /build <nixDockerImage>@sha256<imageSHA256> /bin/sh
 ```
 
 Command notes:
-- Mounts a shared volume from the project's root with a new folder in the container, `/build`, allowing for the contents of the container to be shared with the host and vice verse
-- `nixDockerImage` is a Docker image that has the `nix` commands preinstalled. [This](https://hub.docker.com/r/nixos/nix) is one such image that can be used.
+- Mounts a shared volume from the project's root with a new folder in the container, `/build`, allowing for the contents
+  of the container to be shared with the host and vice verse
+- `nixDockerImage` is a Docker image that has the `nix` commands preinstalled. [This](https://hub.docker.com/r/nixos/nix)
+  is one such image that can be used.
 - `imageSHA256` is the Docker image's digest, which safely pins the image to a specific version.
 
+## Using the overlay
+
+All packages defined in `overlay.nix` can be used as a [nix overlay](https://nixos.org/manual/nixpkgs/stable/#chap-overlays)
```

Comment:
So if I am understanding this correctly, this is like a docker image layers where you can layer packages onto one another to combine them all into one package. The big difference is that you can control if you want a package defined in a lower layer to override what was already put in the overlay? It also allows for packages to _reuse_ dependencies that can be shared between packages?

Reply:
It allows us to incorporate our own pkgs with the nixpkg tree or override specific existing pkgs in nixpkgs with custom configuration. Check out the wiki on this: https://nixos.wiki/wiki/Overlays for some other examples.

But the main idea is reusability. You can add this nix-garage overlay to your local nix environment and install things just like you would anything from the upstream nixpkg repo. Where this gets really cool is when we need to mix in N overlays from various sources 🙂  but well get to that later

Reply:
Ahhhh I see it's basically "inheriting" the base nixpkgs set, but adding in our own. That's really cool reminds me a lot of inheritance in OOP (I like those languages ahaha 😆 ). Except this sounds a lot more flexible than inheritance, so I'm all in for that. I'll have to read up more on overlays!